drivers/soundwire/intel.c

```diff
@@ -1344,17 +1344,18 @@ static int intel_init(struct sdw_intel *sdw)
 {
 	bool clock_stop;
 
+	dev_err(sdw->cdns.dev, "%s: start\n", __func__);
+
 	/* Initialize shim and controller */
 	intel_link_power_up(sdw);
 
 	clock_stop = sdw_cdns_is_clock_stop(&sdw->cdns);
 
-	intel_shim_init(sdw, clock_stop);
+	dev_err(sdw->cdns.dev, "%s: shim_init clock stop %d\n", __func__, clock_stop);
 
-	if (clock_stop)
-		return 0;
+	intel_shim_init(sdw, clock_stop);
 
-	return sdw_cdns_init(&sdw->cdns, false);
+	return 0;
```

Comment:
yes, it's poorly explained. The point is that we need to call sdw_cdns_init() with the multi-master argument reflecting what the module parameter is. This is a left-over from the previous implementation which results in first initializating in multi-master==false, then doing a reinitialization later with multi-master== true (if enabled).

Reply:
Thanks for the explanation. In short, we don't want to call sdw_cdns_init() until we know the value of multi-master, right?

Reply:
it's more that we want to to the Cadence init after we call sync_arm. there's apparently a recommended programming sequence to avoid lockups when enabling the IOs during the inits